### PR TITLE
fix: user secret support

### PR DESCRIPTION
**Pull request by richm**
Cause: The management of secrets was not taking the user into account.

Consequence: Secrets could only be created for the root user.

Fix: Add support for the `podman_run_as_user` parameter, and a
per-secret `run_as_user`, to the secrets management tasks.

Result: Users can create secrets for non-root and root users.
Users can specify a different user for each secret.

Signed-off-by: Rich Megginson <rmeggins@redhat.com>

### CODE REVIEW

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

File: README.md [hunk 1]
```diff
@@ -195,8 +195,15 @@ quadlet container unit spec on the managed node.
...
 
-This is a list of secret specs in the same format as used by
+This is a list of secret specs in almost the same format as used by
 [podman_secret](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_secret_module.html#ansible-collections-containers-podman-podman-secret-module)
+There is an additional field:
+
+* `run_as_user` - Use this to specify a secret for a specific user.  If you do
+  not specify this, then the global default `podman_run_as_user` value will be
+  used. Otherwise, `root` will be used.  NOTE: The user must already exist - the
+  role will not create.
```
Comment (vrothberg) [inline README.md:205]:
```suggestion
  role will not create one.
```
?

**Comment by richm** [comment]
*NOTE* - el8 and fedora quadlet_basic test will fail - there is apparently a bug with podman 4.6.0-1 with rootless quadlet containers.

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Comment by richm** [comment]
[citest]

**Pull request merged**